tests/integration/features/affinity/02-psmdb.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: kuttl.dev/v1
+kind: TestStep
+---
+apiVersion: everest.percona.com/v1alpha1
+kind: DatabaseCluster
+metadata:
+  name: test-psmdb-cluster
+  finalizers:
+  - everest.percona.com/upstream-cluster-cleanup
+  - foregroundDeletion
+spec:
+  engine:
+    type: psmdb
+    version: 7.0.12-7
+    config: |
+     operationProfiling:
+        mode: slowOp
+    userSecretsName: psmdb-sample-secrets
+    replicas: 3
+    storage:
+      size: 15G
+    resources:
+      cpu: 300m
+      memory: 500M
+    affinity:
+      nodeAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 5
+          preference: 
+            matchExpressions:
+            - key: another-node-label-key
+              operator: In
+              values:
+              - another-node-label-value
+  proxy:
+    type: mongos
+    replicas: 3
+    affinity:
+      nodeAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 5
+          preference: 
+            matchExpressions:
+            - key: another-node-label-key
+              operator: In
+              values:
+              - another-node-label-value
```

Comment:
Can we change this to a sharded cluster and set the cfgsrv affinity so that we also cover that case?

Reply:
updated